sonar-php-plugin/src/main/java/org/sonar/plugins/php/PhpPlugin.java

```diff
@@ -106,6 +107,20 @@ public void define(Context context) {
           .subCategory(PHPUNIT_SUBCATEGORY)
           .build());
     }
+    if (context.getSonarQubeVersion().isGreaterThanOrEqual(SQ_VERSION_6_2)) {
```

Comment:
Yes, I had started doing it, then over-engineered it, then dropped because too complex. Your JS version is simple though, I've now done something similar.